src/main/java/io/jenkins/plugins/ctrlplane/api/JobAgent.java

```diff
@@ -0,0 +1,541 @@
+        }
+
+        // Add system/environment info if desired
+        // config.put("os", System.getProperty("os.name"));
+        // config.put("arch", System.getProperty("os.arch"));
+
+        return config;
+    }
+
+    /**
+     * Gets the agent ID as a string if the agent is registered.
+     *
+     * @return the agent ID as a string, or null if not registered
+     */
+    public String getAgentId() {
+        return agentIdRef.get();
+    }
+
+    /**
+     * Gets the next jobs for this agent.
+     *
+     * @return a list of jobs (represented as Maps), empty if none are available or if the agent is not registered
+     */
+    public List<Map<String, Object>> getNextJobs() {
+        String agentId = agentIdRef.get();
+        if (agentId == null) {
+            // ensureRegistered will now attempt registration AND set the ID if successful.
+            if (!ensureRegistered()) {
+                // ensureRegistered logs the specific error (request failed or no ID in response)
+                LOGGER.error("Cannot get jobs: Agent registration/upsert failed or did not provide an ID.");
+                return Collections.emptyList();
+            }
+            // Re-check if agentId was set by ensureRegistered
+            agentId = agentIdRef.get();
+            if (agentId == null) {
+                // This condition should technically not be reachable if ensureRegistered returned true,
+                // as true implies the agentIdRef was set. Log an internal error if it happens.
+                LOGGER.error(
+                        "Internal error: ensureRegistered returned true but agent ID is still null. Cannot get jobs.");
+                return Collections.emptyList();
+            }
+        }
+
+        // Use the correct endpoint from the API spec: /v1/job-agents/{agentId}/queue/next
+        String path = String.format("/v1/job-agents/%s/queue/next", agentId);
+
+        // The response structure is different - it has a "jobs" property containing the array
+        Map<String, Object> response = makeHttpRequest("GET", path, null, new TypeReference<Map<String, Object>>() {});
+
+        if (response != null && response.containsKey("jobs")) {
+            try {
+                @SuppressWarnings("unchecked")
+                List<Map<String, Object>> jobs = (List<Map<String, Object>>) response.get("jobs");
+                LOGGER.debug("Successfully fetched {} jobs from Ctrlplane for agent: {}", jobs.size(), agentId);
+                return jobs;
+            } catch (ClassCastException e) {
+                LOGGER.error("Unexpected response format from jobs endpoint: {}", e.getMessage());
+                return Collections.emptyList();
+            }
+        } else {
+            LOGGER.warn("Failed to fetch jobs or no jobs available for agent: {}", agentId);
+            return Collections.emptyList(); // Return empty list on failure or empty response
+        }
```

Comment:
```suggestion
        if (response != null && response.containsKey("jobs")) {
            try {
                @SuppressWarnings("unchecked")
                List<Map<String, Object>> jobs = (List<Map<String, Object>>) response.get("jobs");
                LOGGER.debug("Successfully fetched {} jobs from Ctrlplane for agent: {}", jobs.size(), agentId);
                return jobs;
            } catch (ClassCastException e) {
                LOGGER.error("Unexpected response format from jobs endpoint: {}", e.getMessage());
                return Collections.emptyList();
            }
        } else {
            LOGGER.warn("Failed to fetch jobs or no jobs available for agent: {}", agentId);
            return Collections.emptyList(); // Return empty list on failure or empty response
        }
```

retunr early and reduce nesting